snow/consensus/snowstorm/conflicts/conflicts.go

```diff
@@ -0,0 +1,195 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package conflicts
+
+import (
+	"errors"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/choices"
+	"github.com/ava-labs/avalanchego/snow/events"
+)
+
+var (
+	errInvalidTxType = errors.New("invalid tx type")
+)
+
+type Conflicts struct {
+	// track the currently processing txs
+	txs map[[32]byte]Tx
+
+	// track which txs are currently consuming which utxos
+	utxos map[[32]byte]ids.Set
+
+	// keeps track of whether dependencies have been accepted
+	pendingAccept events.Blocker
+
+	// keeps track of whether dependencies have been rejected
+	pendingReject events.Blocker
+
+	// track txs that have been marked as ready to accept
+	acceptable []choices.Decidable
+
+	// track txs that have been marked as ready to reject
+	rejectable []choices.Decidable
+}
+
+func New() *Conflicts {
+	return &Conflicts{
+		txs:   make(map[[32]byte]Tx),
+		utxos: make(map[[32]byte]ids.Set),
+	}
+}
+
+// Add this tx to the conflict set. If this tx is of the correct type, this tx
+// will be added to the set of processing txs. It is assumed this tx wasn't
+// already processing. This will mark the consumed utxos and register a rejector
+// that will be notified if a dependency of this tx was rejected.
+func (c *Conflicts) Add(txIntf choices.Decidable) error {
```

Comment:
Is there any reason not to make the parameter type Tx if we're going to do the type assertion immediately?